application-diagram-api/src/main/java/com/xwiki/diagram/internal/DiagramLinksListener.java

```diff
@@ -127,38 +109,35 @@ public Object doInHibernate(Session session) throws XWikiException
                         // Is necessary to blank links from doc.
                         context.remove("links");
 
-                        // Get pages included by this diagram.
-                        for (String includedPage : contentHandler.getIncludedPages(document.getContent())) {
-                            XWikiLink wikiLink = getXWikiLink(document.getId(),
-                                localEntityReferenceSerializer.serialize(document.getDocumentReference()),
-                                resolver.resolve(includedPage));
+                        // Get pages linked by this diagram.
+                        for (DocumentReference linkedDocRef : contentHandler.getLinkedPages(document.getContent())) {
```

Comment:
I see that you pass only the document content and you resolve the found relative references using the current document reference. Does it mean that the current document is always the diagram? It feels a bit fragile. I would pass the diagram reference along with the diagram content, if it's not too complex.